aspnetcore/mvc/views/tag-helpers/built-in/AnchorTagHelper.md

```diff
@@ -34,35 +32,110 @@ The speaker controller used in attribute definitions below is shown here.
+
+
+```html
+<a href="/Home">All Speakers</a>
+```
 
 - - -
   
 ### asp-action
 
-`asp-action` is the name of the method in the controller that will be included in the final URL. That is, in the example, if the route to the Speaker Detail page is wanted, then the attribute should be set to `asp-action=Detail`. You should always set `asp-controller` when specifying `asp-action`. If no `asp-action` is specified then the default `asp-controller` will be the current executing controller.
+`asp-action` is the name of the action method in the controller that will be included in the generated `href`. For example, the following code set the generated `href` to point to the speaker detail page:
+
+```html
+<a asp-controller="Speaker" asp-action="Detail" >Speaker Detail</a>
+```
+
+The generated URL will be
+
+```html
+<a href="/Speaker/Detail">Speaker Detail</a>
+```
+
+If no `asp-controller` attribute is specified, the default controller calling the view executing the current view will be used.  
+ 
+If the attribute `asp-action` is `Index`, then no action is appended to the URL, leading to the default `Index` method being called. The action specified (or defaulted), must exist in the controller referenced in `asp-controller`.
 
 - - -
   
 ### asp-route-{value}
 
-`asp-route-` is a wild card route prefix. Any value you put after the trailing dash will be interpreted as the parameter to pass into the route. For example, if a tag is created as follows: 
+`asp-route-` is a wild card route prefix. Any value you put after the trailing dash will be interpreted as a potential route parameter. If a default route is not found, this route prefix will be appended to the generated href as a request parameter and value.  Otherwise it will be substituted in the route template.
+
+Assuming you have a controller method defined as follows:
+
+```csharp
+public IActionResult AnchorTagHelper(string id)
+{
+    var speaker = new SpeakerData()
+    {
+        SpeakerId = 12
+    };      
+    return View(viewName, speaker);
+}
+```
 
-`<a  asp-controller="Speaker" asp-action="Detail" asp-route-id-="11">Speaker 11</a>`
+and have the default route template defined in your **startup.cs** as follows:
 
-the `href` generated will be 
+```csharp
+app.UseMvc(routes =>
+{
+   routes.MapRoute(
+    name: "default",
+    template: "{controller=Home}/{action=Index}/{id?}");
+});
 
-`<a href="/Speaker/11">Speaker 11</a>` 
+```
 
-This is because a route was found that matched a single parameter "id" in the ```SpeakerController``` method ```Detail```. If there was no parameter match, say for example you created the tag helper 
+The **cshtml** file that contains the Anchor Tag Helper necessary to use the **speaker** model parameter passed in from the controller to the view is as follows:
```

Comment:
no style on cshtml 

`speaker` should be code fenced, not bold. Do you ever show the speaker model?